ch4/ofi does not build correctly on mac os with `--disable-shared`

I see the following error when trying to build `ch4/ofi` on mac os with `--disable-shared`:

```
Undefined symbols for architecture x86_64:
  "_MPIDI_CH4_SHM_POSIX_FBOX_GENERAL", referenced from:
Undefined symbols for architecture x86_64:
  "_MPIDI_CH4_SHM_POSIX_FBOX_GENERAL", referenced from:
      _MPIDI_POSIX_mpi_init_hook in libpmpi.a(lib_libpmpi_la-posix_init.o)
      _MPIDI_POSIX_mpi_init_hook in libpmpi.a(lib_libpmpi_la-posix_init.o)
  "_MPIDI_POSIX_eager_fbox_control_global", referenced from:
  "_MPIDI_POSIX_eager_fbox_control_global", referenced from:
      _arb_progress_hook in libpmpi.a(lib_libpmpi_la-ch4_init.o)
      _MPIDI_anysource_matched in libpmpi.a(lib_libpmpi_la-ch4_init.o)
      _MPIDI_OFI_handle_lmt_ack in libpmpi.a(lib_libpmpi_la-ch4_init.o)
      _MPIDI_OFI_do_handle_long_am in libpmpi.a(lib_libpmpi_la-ch4_init.o)
      _arb_progress_hook in libpmpi.a(lib_libpmpi_la-ch4_init.o)
      _MPIDI_anysource_matched in libpmpi.a(lib_libpmpi_la-ch4_init.o)
      _MPIDI_OFI_handle_lmt_ack in libpmpi.a(lib_libpmpi_la-ch4_init.o)
      _MPIDI_OFI_do_handle_long_am in libpmpi.a(lib_libpmpi_la-ch4_init.o)
      _progress_fn in libpmpi.a(lib_libpmpi_la-async.o)
      _MPIDIG_do_irecv in libpmpi.a(lib_libpmpi_la-async.o)
      _MPIDIG_handle_unexpected in libpmpi.a(lib_libpmpi_la-async.o)
      _progress_fn in libpmpi.a(lib_libpmpi_la-async.o)
      _MPIDIG_do_irecv in libpmpi.a(lib_libpmpi_la-async.o)
      _MPIDIG_handle_unexpected in libpmpi.a(lib_libpmpi_la-async.o)
      ...
      ...
```

Here's the full config line that I used, but the problem seems to be with `--disable-shared` (the error goes away if I don't disable shared libraries):

```
../mpich/configure --prefix=/Users/balaji/software/mpich/build/install --disable-romio
    --enable-g=all --enable-spawn --enable-maintainer-mode --enable-cxx
    --with-device=ch4:ofi --disable-shared CC=clang CXX=clang++ FC=gfortran
    --enable-strict=noopt --disable-fast --disable-perftest
```

--- a/src/mpid/ch4/shm/posix/eager/fbox/globals.c
+++ b/src/mpid/ch4/shm/posix/eager/fbox/globals.c
@@ -12,9 +12,10 @@
 #include "fbox_impl.h"
 #include "fbox_types.h"
 
-/* *INDENT-OFF* */
-MPIDI_POSIX_eager_fbox_control_t MPIDI_POSIX_eager_fbox_control_global;
-/* *INDENT-ON* */
+/* Note: without the following initialiazation, the common symbol may get lost
+   during linking. This is manifested by static linking on mac osx.
+*/
+MPIDI_POSIX_eager_fbox_control_t MPIDI_POSIX_eager_fbox_control_global = { 0 };
 
 #ifdef MPL_USE_DBG_LOGGING
 MPL_dbg_class MPIDI_CH4_SHM_POSIX_FBOX_GENERAL;

